### ISSUE TITLE
2019/6/4 - 2019/8/22

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#99]
1.Q8
2.Q9
3.Q10
4.Neural Machine Translation
5.1 - Translating human readable dates into machine readable dates
6.1.1 - Dataset
7.2 - Neural machine translation with attention
8.2.1 - Attention mechanism
9.one_step_attention
10.----------------------------------------------------------------------------------------------------------------------
11.model
12.Trigger word detection
13.1 - Data synthesis: Creating a speech dataset
14.1.1 - Listening to the data
15.1.2 - From audio recordings to spectrograms
16.1.3 - Generating a single training example
17.is_overlapping
18.insert_audio_clip
19.insert_ones(y, segment_end_ms) inserts 1's into our label vector y after the word "activate"
20.----------------------------------------------------------------------------------------------------------------------
21.y -- numpy array of shape (1, Ty), the labels of the training example
22.What is segment_end_y? 
23.insert_ones
24.create_training_example
25.1.4 - Full training set
26.1.5 - Development set
27.2 - Model
28.2.1 - Build the model
29.model
30.----------------------------------------------------------------------------------------------------------------------
31.2.2 - Fit the model
32.2.3 - Test the model
33.3 - Making Predictions
34.3.3 - Test on dev examples
35.Buy a bag; 
36.Rent a house; Find location of zisong; cannot log on the web; (2019-6-4)
37.Deal with hukou migration; check on zhihu about migration; need luqutongzhishu; (2019-6-4)
38.Review all assignment; download files from github; done; (2019-6-4)
39.5 - Backpropagation in convolutional neural networks (OPTIONAL / UNGRADED)
40.----------------------------------------------------------------------------------------------------------------------
41.5.1 - Convolutional layer backward pass
42.5.1.1 - Computing dA:
43.5.1.2 - Computing dW:
44.5.1.3 - Computing db:
45.5.2 Pooling layer - backward pass
46.5.2.1 Max pooling - backward pass
47.5.2.2 - Average pooling - backward pass
48.Select a mosquito nets; done; (2019-6-5)
49.【求助】在哪些网站可以获取附近的租房信息呢？白云黄鹤bbs不知怎么回事，我昨天今天都登不上去。
50.----------------------------------------------------------------------------------------------------------------------
51.Book the ticket; rent the house; (2019-6-7)
52.List: 
53.身份证；银行卡；
54.档案夹；手表；U盘；耳机；眼镜盒；卫生纸；充电宝；剃须刀；铅笔盒；手机架；电动牙刷；(2019-6-10)
55.Ask Xu location of lab; (2019-6-10)
56.Talk with Prof He to deal with my direction; (2019-6-11)
57.What is AI 芯片化; not to do this; (2019-6-11)
58.What is 边缘计算; not to do this; (2019-6-11)
59.Get to know Verilog HDL; (2019-6-11)
60.----------------------------------------------------------------------------------------------------------------------
61.Verilog HDL就是为开发FPGA而开发的一款工具而已，就像是盖房子用的斧子锤子一样，要多动手才能精通; (2019-6-11)
62.Check what is modelsim; done; (2019-6-11)
63.Learn where to buy a bicycle; find bbs of second hand; give up; (2019-6-12)
64.功能仿真和时序仿真；(2019-6-12)
65.What is PFGA 仿真；done; we can start from an adder; (2019-6-12)
66.HDL 顾名思义Hardware Description Languag; find some examples of verilog HDL; (2019-6-12)
67.加法器; 比较器; 三态驱动器; 全加器(一位全加器，两位全加器); 
68.Check operations in linear regression; check verilog matrix; done; (2019-6-12)
69.Check for versions of quartus ii; give up; (2019-6-12)
70.----------------------------------------------------------------------------------------------------------------------
71.Try to download and install quartus ii; done; try to install devices; (2019-6-12)
72.Check zhihu about quartus ii; give up; (2019-6-12)
73.Back up database; done; (2019-6-12)
74.Download modelsim; install; deal with environment variables; done; (2019-6-12)
75.Try to neglect (0) for noun; we can use () for verb and do nothing for noun; done; (2019-6-12)
76. Try to build an adder; not succeed; try to search by English; write a schematic file; compile; (2019-6-12)
77. Figure out where to buy fruit; done; (2019-6-12)
78.Try to see my previous machine learning code by octave; we should try to use data of Andrew Wu, since I cannot come up with new data; (2019-6-13)
79.Enroll in a new session; (2019-6-13)
80.----------------------------------------------------------------------------------------------------------------------
81.Learn to start python; learn how to write python in a file; (2019-6-13)
82.Buy an bicycle; go to jimao; buy scissors and another wires; (2019-6-13)
83.Read txt to numpy array; learn to use python to plot; learn to plot array; (2019-6-14)
84.Matplotlib draw straight line; hold; (2019-6-14)
85.Read through ex1; we use same file name as matlab; done; (2019-6-14)
86.Get m; data type not understood; combine vectors by column; np.c_[np.ones([m, 1]), a[:, 0]]; (2019-6-14)
87.Learn to call functions by python; define function in the same file; done; define function in another file; done; return from parameters; done; (2019-6-14)
88.Write compute cost; define theta; change a to data; done; add comments to python; done; (2019-6-14)
89.Use python to create 2D array; we should pay attention to its format; (2019-6-14)
90.----------------------------------------------------------------------------------------------------------------------
91.Deal with gradient descent; strange, result is different; it is OK; (2019-6-14)
92.Draw dots and line; change color; done; (2019-6-14)
93.Deal with logic regression; run octave code to see result; 
94.Replace , with ; done; load data; done; (2019-6-15)
95.What is plotData(X, y); first let’s plot all in the same type; done; learn how to find indices for python; numpy.where(x == 0)[0]; (2019-6-15)
96.Separate plotData; done; set label; done; add legend; done; (2019-6-15)
97.Write sigmoid; done; (2019-6-15)
98.Write cost function; learn how to return multiple; done; check log; done; (2019-6-15)
99.fminunc alternate in numpy; hold; use gradient descent to find minimum; 
100.----------------------------------------------------------------------------------------------------------------------
101.Read an article about logistic regression in python; done; do we need to separate cost and grad? Yes; separate cost and gradient; change cost function to computeCost; done; (2019-6-15)
102.What is initial theta? Done; change name of copy function; done; check our cost function with copied function; change Sigmoid to lower case; done; check out gradient with with copied ones; done; (2019-6-15)
103.Python 分号; delete; done; (2019-6-15)
104.Plot boundary; give up; learn how to set jingdu; use round; done; check whether two gradient is same; yes, let us change them to hypothesis; done; (2019-6-16)
105.Go through ex2_reg; (2019-6-16)
106.Deal with mapFeature; hold; (2019-6-16)
107.Let us see the result of neural network of octave; done; how to load mat with python; we still need neural network; see backprop algorithm; (2019-6-16)
108.Gradient descent => NN = > RNN
109.那么深层网络和浅层网络相比有什么优势呢？简单来说深层网络能够表达力更强。
110.----------------------------------------------------------------------------------------------------------------------
111.为了理解神经网络，我们应该先理解神经网络的组成单元——神经元。神经元也叫做感知器。
112.例子：用感知器实现and函数
113.感知器的训练
114.我们使用一个可导的线性函数来替代感知器的阶跃函数
115.比如我们可以用和的差的平方的来表示它们的接近程度
116.按照上面的讨论，我们就可以写出梯度下降算法的公式
117.因为我们已经写了感知器的代码，因此我们先比较一下感知器模型和线性单元模型，看看哪些代码能够复用。
118.神经元和感知器本质上是一样的，只不过我们说感知器的时候，它的激活函数是阶跃函数；而当我们说神经元时，激活函数往往选择为sigmoid函数或tanh函数。
119.因为我们这个神经网络是全连接网络，所以可以看到每个节点都和上一层的所有节点有连接。
120.----------------------------------------------------------------------------------------------------------------------
121.对于这些人为设置的参数，我们称之为超参数(Hyper-Parameters)
122.对于全连接网络，隐藏层最好不要超过三层。
123.Use python to rewrite NN of Andrew Wu; it it hard to find the code; (2019-6-16)
124.NN那里要绕路，不如直接进攻RNN；we must deal with NN first; 
125.Download MNIST; 
126.入门级解读：小白也能看懂的TensorFlow介绍
127.在 2 个特征的情形中，我们也可以使用线性回归，但是需要创建一个平面（而不是直线）
128.investigate anaconda; done; investigate how to install tensorflow; done; (2019-6-17)
129.investigate jupyter notebook; update conda; print hello world; 
130.----------------------------------------------------------------------------------------------------------------------
131.+ 按钮用于创建新的单元格
132.点击“File”->Rename，可以对notebook文件进行重命名，这里我命名成‘Helloworld’
133.其实Notebook 就是个扩展名为 .ipynb 的大型 JSON 文件
134.点击下面的保存按钮，可以保存你的notebook文件。但 notebook 也会定期自动保存。
135.Use jupyter notebook to write linear regression; done; use jupyter notebook to write logic regression; what is wrong with gradient? Delete sigmoid; done; (2019-6-17)
136.Use tensorflow to implement linear regression; run example tensorflow; 
137.No module named 'tensorflow'; there is no tensorflow package; install tensorflow; check python version and conda version; 
138.直接在install 后面加一个--user就可以了; done; (2019-6-17)
139.Let us try hand-writing recognition; read through; 
140.----------------------------------------------------------------------------------------------------------------------
141.API; 应用程序接口; walk through API of tensorflow; 
142.investigate how many type of session; 
143.TensorFlow是谷歌基于DistBelief进行研发的第二代人工智能学习系统，其命名来源于本身的运行原理。Tsnsor（张量）意味着N维数组，Flow(流)意味着基于数据流图的计算，TensorFlow为张量从流图的一端流动到另一端的计算过程。TensorFlow是将复杂的数据结构传输至人工神经网络中进行分析和处理过程的系统。
144.为了方便使用诸如IPython之类的Python交互环境，可以使用InteractiveSession替代Session类，使用Tensor.eval() 和Operation.run()方法来代替Session.run()，这样可以避免使用一个变量来持有会话
145.可以看到第一个值value是必须的，可以是一个数值，也可以是一个列表。 
146.walk through tensorflow constant; 而如果列表大小小于shape大小，则会用列表的最后一项元素填充剩余的张量元素：
147.tf.global_variables_initializer(); run tensorflow variables; 
148.sess.run(v.initializer)     #运行变量的initializer。调用op之前，所有变量都应被显式地初始化过。
149.那么在低级别API里面，最简单的方法就是添加一个给所有变量初始化的操作（比如tf.global_variables_initializer()），并在使用模型之前首先运行那个操作。
150.----------------------------------------------------------------------------------------------------------------------
151.print(sess.run(tf.report_uninitialized_variables())); done; (2019-6-18)
152.with tf.Session() as sess: to replace original; done; The tf. operator performs the specific operation and the output will be a Tensor. The attribute name="some_name" is defined for better Tensorboard visualization as we see later in this tutorial.; (2019-6-18)
153.Use tensorflow to implement linear regression; run sample; set initial value as zero; done; (2019-6-18)
154.Read train_X and train_Y from txt; done; figure out functions in linear regression; 
155.tf.multiply（）两个矩阵中对应元素各自相乘; tf.matmul（）将矩阵a乘以矩阵b，生成a * b。
156.reduce_sum应该理解为压缩求和，用于降维; 给定一个张量 x 和一个张量 y,这个操作计算对应于 \(x^y\)  中的 x 元素和 y 元素.
157.What is feed_dict? feed_dict就是用来赋值的，格式为字典型。比如，对上一章节中的a进行赋值。有feed_dict={a:8}。首先占位符申请空间；使用的时候，通过占位符“喂（feed）”给程序。然后程序就可以run了。。。
158.Read another version of linear regression; 
159.To make it easier to understand, debug, and optimize TensorFlow programs, we've included a suite of visualization tools called TensorBoard.
160.----------------------------------------------------------------------------------------------------------------------
161.The tf. operator performs the specific operation and the output will be a Tensor. The attribute name="some_name" is defined for better Tensorboard visualization as we see later in this tutorial.
162.Replace constant with feed_dict; 
163.By using tf.variables_initializer, we can explicitly command the TensorFlow to only initialize a certain variable. The script is as follows
164.All variables can be initialized at once using the tf.global_variables_initializer(). This op must be run after the model constructed. 
165.Let us use another way to implement session; yes, according to GeeksforGeeks, this is correct; done; (2019-6-19)
166.Implement logic regression; 
167.这个问题的难度会不会太大了？七大数学难题；
168.组合最优化，在应用数学和理论计算机科学的领域中，组合优化是在一个有限的对象集中找出最优对象的一类课题。在很多组合优化的问题中，穷举搜索／枚举法是不可行的。组合优化的问题的特征是可行解的集是离散或者可以简化到离散的，并且目标是找到最优解。常见的例子有旅行商问题和最小生成树。二维的例子，比如服装厂做衣服，衣服分成很多块，这些块需要从布料上切下来。怎么切，剩下的废布料最少？三维的例子，如集装优化。
169.模拟退火和遗传算法，都体现了道法自然。(2019-6-19)
170.----------------------------------------------------------------------------------------------------------------------
171.如果做好了所有准备再开始科研，那么永远将无法起步；做一些初步的准备，就可以强行开始了，之后需要什么再去查什么补什么；(2019-6-19)
172.Read zhihu about reinforcement learning; done; (2019-6-19)
173.Read zhihu about pointer network; what is convex hull? 
174.We still need to learn RNN model before we start next section; 
175.Use theta to rewrite linear regression; initialize theta; tensorflow initialize matrix; done; (2019-6-19)
176.Try to find quartus verilog tutorial; give up; (2019-6-19)
177.搞科研很重要的是搞清楚自己所处的位置；当前任务上下左右的知识点是什么；以及之前做的什么，之后打算做什么；这就是搞科研“格局感”；(2019-6-15)
178.之所以不选强化学习，是因为那个在我的掌控范围之外；奖励函数的随机性，和化学实验的调温度实在有异曲同工之妙；同时自然语言处理也和实际贴合更近一点，比那些玩超级玛丽的更接近于工程；(2019-6-19)
179.Write logistic regression; 
180.----------------------------------------------------------------------------------------------------------------------
181.Do mid variables need declaration? No, pred is an example; (2019-6-20)
182.Use -1 is very convenient; initialize X, Y and theta; try to use tf.zeros, done; (2019-6-20)
183.bad operand type for unary -: 'Dimension'; solved; (2019-6-20)
184.The problem is how to deal with matrix; let us rewrite linear regression as matrix; use another way to write zip; done; (2019-6-20)
185.Can we use matrix to run optimizer? Done; (2019-6-20) 
186.Strange, cost is nan; use tensorflow’s way to write cost; done; why cost still equals to nan? You are overflowing float32 because the learning rate is too high for your problem, and instead of converging the weight variable (W) is oscillating towards larger and larger magnitudes on each step of gradient descent; (2019-6-20)
187.Do we have better optimize method? Give up; (2019-6-20)
188.Let’s see RNN model; search zhihu about RNN; 
189.一文搞懂RNN（循环神经网络）基础篇; 循环神经网络的隐藏层的值s不仅仅取决于当前这次的输入x，还取决于上一次隐藏层的值s。权重矩阵 W就是隐藏层上一次的值作为这一次的输入的权重。
190.----------------------------------------------------------------------------------------------------------------------
191.Build a simple RNN model; figure out size of W; W.shape=(prev_layer+1, current_layer); (2019-6-20)
192.Review ex4 of machine learning; let us display theta1 and theta2; octave print array; fprintf('%d', Theta2); done; (2019-6-20)
193.Random initialize; see y; done; see X; so many data; done; (2019-6-20)
194.Learn how to use modelsim and quartus ii together; give up; (2019-6-20)
195.How octave load data? Let us save X and Y in txt file; print matrix size; done; try to save matrix in txt file; done; try to read X and Y from txt data; done; (2019-6-20)
196.Try to load data from python; done; try to save Theta1 and Theta2 to txt; read them from txt to python, done; lambda is a function in python; (2019-6-20)
197.Write nncostfunction; for linear regression, try to separate function; give up; first translate to numpy, then to tensorflow; write our own sigmoid; done; change float matrix to int; done; (2019-6-20)
198.Yes, cost is correct without regularization; deal with the range in python for loop; book a short online; book a mosquito killer; range contain the floor, but does not contain ceiling; (2019-6-20)
199.Cost is correct with regularization; let us first implement by numpy, then by tensorflow; as we can see, optimize needs compute gradient; Optimization and Root Finding (scipy.optimize); Initial guess. Array of real elements of size (n,), where ‘n’ is the number of independent variables; Extra arguments passed to the objective function and its derivatives (fun, jac and hess functions); (2019-6-21)
200.----------------------------------------------------------------------------------------------------------------------
201.Use optimize minimize to write linear regression; write function of computeGradient; done; theta must be the first variable; (2019-6-21)
202.Delete extra reshape; done; delete gradient descent; done; (2019-6-21)
203.Add input layer size to compute cost; use numpy to unroll matrix; theta1.flatten; done; learn to combine two numpy arrays; nn_params = np.concatenate((Theta1_flat, Theta2_flat)); done; use nn_params to write cost function; done; (2019-6-21)
204.Add num_labels to cost function; done; add input layer size and hidden layer size to cost function; done; add nn_params to cost function; done; (2019-6-21)
205.Write sigmoidGradient; done; try to change random function to see the difference of result; it seems that random initialization is important, since the result change quite a lot if we change the functions; done; write rand initialize weight; done; (2019-6-21)
206.Rewrite random initialize weights; done; (2019-6-21)
207.What is checkNNGradient? Write checkNNGradient; 
208.Write debug initialize weights; run to see what the fuck it is; write debug initialize weights; done; (2019-6-21)
209.What the fuck is mod? How to get mod in python; done; what the fuck is @p? it is optimize function; write computeGradient; let us try the strange matrix multiplication of numpy; one dimension array cannot be done transform; do we have other way to express reshape? Give up; (2019-6-22)
210.----------------------------------------------------------------------------------------------------------------------
211.Compute grad and numgrad; let us separate the target function; done; what is compute numerical gradient? Compute cost; done; write compute numerical gradient; change octave p to i; done; combine two columns; give up; what is norm in octave? Use it in numpy; done; (2019-6-22)
212.The second two column is different; why octave result is different from python? Compare theta1 and theta2; theta1 is different; problem is in debugInitializeWeights; it seems that octave allocate according to the column; let us use another way to allocate i; done; (2019-6-22)
213.Check theta1; theta1 is correct, now; check theta2; theta2 is correct; check X; X is correct; check y; done; check nn_params, wrong; the key is flatten(‘F’); done; check grad; grad become correct; does grad has something to do with lambda? Yes, rewrite compute gradient; done; (2019-6-22)
214.check computeNumericalGradient; write computeCost; done; copy back to original; done; deal with cost; add flatten(‘F’), done; 
215.It = 1; 0.32068323946855815；It =6; 0.3196604712926606
216.Let us try max iteration on linear regression; write predict; what is pred? 
217.0, 0；1, 10；2, 10；10, 9.88; finally; (2019-6-22)
218.Can we read mat file? Yes; try to load Theta1 and Theta2 from mat file; done; read X and Y from mat file; done; delete extra file; done; (2019-6-23)
219.Can we use our neural network to deal with mnist? Mnist is also handwriting; let us try to use train set in tensorflow; OK; we can see that every time the result is different; done; what is test set and validation set? Validation set deal with hyperparameters; done; (2019-6-23)
220.----------------------------------------------------------------------------------------------------------------------
221.Read data from MNIST data; can we store them in a numpy array? Investigate the appearance of X and y; done; copy relative data from python to tensorflow; deal with load data; done; success; delete extra comments; done; (2019-6-23)
222.To evaluate it, we had to run init=tf.global_variables_initializer(); To summarize, remember to initialize your variables, create a session and run the operations inside the session;  A placeholder is an object whose value you can specify only later. To specify values for a placeholder, you can pass in values by using a "feed dictionary" (feed_dict variable); the second method does not need to close session; (2019-6-23)
223.In my handwriting, change train set to test set in the predict function; change X and Y from train set; change X and Y to X_train and Y_train; done; change train set to test set in the predict function; change it be more accurate if we raise the iteration? Yes, it is more accurate now; (2019-6-24)
224.Write progress report to Prof He; done; read through tensorflow of Andrew Ng; this article shows us the function one by one; this article also tells us we can place tensorflow in a function; num_px * num_px = 64 * 64 * 3 = 12288; tf.contrib.layers.xavier_initializer; W1 and b1 are both combined in theta1; is the cost really equals to our own? (2019-6-24)
225.Separate W and b is our hand-writing program; strange; we must modify X_train; change X_train to X_test; it becomes right; let us change back; change iteration to 60; okay, done; let us first work on test set, it is better; (2019-6-24)
226.ops.reset_default_graph(); give up; tf.set_random_seed; Sets the graph-level random seed for the default graph; it seems that write a new is easier than modify an old; the program reports that No module named 'tf_utils'; check the functions in tf_utils; load_dataset; we can write our own; random_mini_batches; hold; convert_to_one_hot; we can write our own; predict; we can write our own; what is h5py; hold; (2019-6-24)
227.Let us try previous function of tensorflow; can we change profile in coursera? Give up; we can see that coursera is transpose of our matrix; we should first load the data; done; if predict accuracy is higher than 10%, then it works; do we need to transpose X and Y? As we can see, they print after finishing loading; yes, transpose; done; (2019-6-24)
228.Deal with forward prop; done; deal with cost; done; deal with model; deal with minibatches; done; we should place accuracy in the iteration; change learning rate to see whether we can get better result; obviously wrong, the rate should be higher than 10; let us try sigmoid; yes, it become larger than 10; change hidden layer size; give up; (2019-6-24)
229.Buy an remote control; review neural network in machine learning; it is important to know the shape of all function; find the how to do research pdf; read guide of MIT; get really a lot; buy socks; done; review gradient explode and gradient vanish; read MIT graduate entrance guide; 
230.----------------------------------------------------------------------------------------------------------------------
231.Change to limitations and future work; (Limitation) The prerequisite that we can do optimize to a problem is that there are many solution to this problem. As a result, SC works better on engineering problem than mathematical problems which have only one answer. (2019-6-25)
232.  To see the sentence of limitations and future work; download all papers recommended by Prof. He; let us start our RNN model; Wax, Waa, Wya, ba, by are all in parameters; WTF is softmax; review softmax in the past; use python to implement softmatx; done; go through the dinosaur name; (2019-6-26)
233.Create dinosaur name txt book; rename downloaded papers; done; find read paper method from peng jianming; what is softmax function? Its dimension is equal, also K; logic function is used to describe the S figure; the reason why we use exponent is to make sure that each value is positive; continue on dinosaur; (2019-6-30)
234.There is a difference in function rnn_forward between rnn model and dinosaurs; what is a? as we can see, they name variables using dimensions; replace cache in RNN model; it seems that they use cache to check the result; delete cache in RNN; done; delete parameters; give up; (2019-6-30)
235.Run backprop of RNN; read article about backprop; too hard; review simple backprop; review logic regression; done; modify note of machine learning; done; review neural network; logic regression is classification; logic regression is to choose theta; what is h(theta) in logistic regression; np.exp is exponent; cost function only contain last column, namely output; (2019-7-1)
236.X, Y = create_placeholders(n_x, n_y)
237.parameters = initialize_parameters()
238.Z3 = forward_propagation(X, parameters)
239.cost = compute_cost(Z3, Y)
240.----------------------------------------------------------------------------------------------------------------------
241.optimizer = tf.train.GradientDescentOptimizer(learning_rate = learning_rate).minimize(cost)
242.init = tf.global_variables_initializer()
243.Try to write backprop of RNN; the shape of hypothesis is really like relu; use 8888 to go to the root; do we need to change compound to sentence; done; can we change file name in github; try to install git on windows; done; how to use vs code and git? Uninstall Quartus II; change elements.html, elements_editor.html, compound.html, compound_editor.html; done; modify navigation tag; done; modify analyzer; done; modify theory.html, done; words fit POS well; (2019-7-1)
244.Change element to word in theory; hold; add school and email to paper; done; what is the content after related work? It is approach; delete overview; done; change last few ids; download mysql workbench; install; not know how to use, give up; (2019-7-1)
245.Implement LSTM following Andrew Ng’s guide; it seems that papers are from junk schools; this is a peer review; what is springer? Public organization? What is ORCIDs? Open Researcher and Contributor ID; (2019-7-2)
246.Modify the experience main page tag; done; modify theory editor main page tag; done; modify experience editor main page tag; done; (2019-7-2)
247.Create phrases.html; done; read our import library code to figure out what is happening; add indent to our own paper; 
248.Change button content of Word Editor; done; change edit_compound to edit_sentence, change edit_element to edit_word; done; vocabulary contains words and phrases; done; download template for papers; done; (2019-7-2)
249.Read through 2017; read CCIS_productionII.docx; done; read contents; done; read corresponding author; done; read organizations; done; read organizations; done; read preface; done; read abstract for invited talks; done; read 8018; done; read bookbackmatter; done; read bookfrontmatter; done; read print.pdf; done; read conf terms; done; read productionII; done; read copyright; done; read guidelines; done; read volume number; done; (2019-7-2)
250.----------------------------------------------------------------------------------------------------------------------
251.Read through 2019; read acceptance note; done; read conf terms; done; read productionII; done; read checklist, done; download copyright form, done; (2019-7-3)
252.Preface和organization是否可以继续沿用之前的？Yes, but modify a little; use tab can make format more beautiful; 
253.Content的页码如何标注？It is assigned by Springer; done; (2019-7-3)
254.Copyright form由每个作者签字？
255.Read proceeding proposal form; get the PC member; (2019-7-3)
256.modify preface; change 2017 to 2019; change place to Lanzhou, Gansu; done; change date to August 2-4; done; change HUST to LZU; done; 今年收了多少文章？今年请谁做演讲？
257.Modify organization; modify general chair; done; modify program chair; delete; modify program co-chair; they have three co-chair this year; done; modify organization chair; done; modify PC member; done; check; done; (2019-7-3)
258.Classify 11 papers into 7 classes; done; check title and author; done; (2019-7-3)
259.(1) Algorithms and Complexity
260.----------------------------------------------------------------------------------------------------------------------
261.Cooperation on the monte carlo rule: Prison’s dilemma game on the grid (r25)
262.(2) Algorithm Designing
263.Special frequency quadrilaterals and an application (r61)
264.(3) Software Theory and Method
265.(4) Data Science and Machine Learning Theory
266.Sampling to maintain approximate probability distribution under chi-square test (r24)
267.Adjusting the Inheritance of Topic for Dynamic Document Clustering (r79)
268.An Improved Proof of the Closure under Homomorphic Inverse of FCFL Valued in Lattice-ordered Monoids (r80)
269.(5) Web Science Base Theory
270.----------------------------------------------------------------------------------------------------------------------
271.Socially-attentive Representation Learning for Cold-Start Fraud Review Detection (r63)
272.(6) Parallel and Distributed Computing
273.Semi-Online Machine Covering on Two Hierarchical Machines with Known Total Size of Low-Hierarchy Jobs (r9)
274.(7)Computational Model: Quantum computing, computational economics, socialcomputing, bioinformatics, cloud computing,
275.A Combined Weighted Concept Lattice for Multi-Source Semantic interoperability of ECG-Ontologies Based on Inclusion Degree and Information Entropy (r3)
276.Minimizing the Spread of Rumor within Budget Constraint in Online Network (r32)
277.Quantum Reversible Fuzzy Grammars (r59); why they keep saying that parameters is wrong? 
278.A Signcryption Scheme Based Learning with Errors Over Rings without Trapdoor (r68)
279.Build a simple LSTM model; not enough values to unpack (expected 10, got 5); caches is generated by lstm_forward; your overall loop structure usually consists of a forward pass, a cost computation, a backward pass, and a parameter update; gradient clip to solve gradient explode; hold on; (2019-7-4)
280.----------------------------------------------------------------------------------------------------------------------
281.Look for keras; replace importlibrary with import ID; syntax first generate symbols, then use physical address to replace the symbol; The name compiler is primarily used for programs that translate source code from a high-level programming language to a lower level language; do preprocessor has name of input and output? No; (2019-7-4)
282.What is &#38; translate source is used in btn_translate; change the button in analyzer.html, done; look for keras first appeared in Andrew Ng’s course; find it; Convolutional Neural Networks/Week 2/Keras Tutorial - The Happy House (not graded); check keras on zhihu; continue Andrew Ng sequence model course week 2 and week 3; make sure the similarity is between 0 and 1; change SelectEntry to SelectSentence; done; rename select_compound as select_sentence; done; change compounds.php to show_all_sentences.php, done; (2019-7-4)
283.modify sort_similarity.php so that similarity value is between 0 and 1; move out sql; done; delete group; we can use word-embedding to implement group; take element_group out of the inner for loop; done; (2019-7-4)
284.Read our translate_source file and understand it; change ElementID to WordID; done; word also fit POS well; delete compiler.php; 
285.The similarity between AB and BA is different, should we change the situation? 
286.Try to run LSTM to make it work; done; upload them; try to modify dinosaur file; what is unique characters; it is about text generation; what is sample; figure out the usage of each function; python set function; example for 26 letters, there is \n; Now assume that your model is trained. You would like to generate new text (characters); y becomes input x through sampling; clip, sample, optimize; stochastic gradient descent; strange, how they get loss from run forward; we must know the function in it; (2019-7-5)
287.What is cross-entropy; 交叉熵损失函数经常用于分类问题中，特别是在神经网络做分类问题时，也经常使用交叉熵作为损失函数，此外，由于交叉熵涉及到计算每个类别的概率，所以交叉熵几乎每次都和softmax函数一起出现; what is convex function; 为什么数学概念中，将凸起的函数称为凹函数; log function is cross-entropy function; check softmax on Zhihu; softmax用于多分类过程中，它将多个神经元的输出，映射到（0,1）区间内，可以看成概率来理解，从而来进行多分类; 用了softmax函数之后，梯度求导过程非常非常方便; we are glad to find util.py, use it to implement out function; success; (2019-7-5)
288.Let us try to change dino.txt to aaa, bbb, ccc and check the result; yes it works; (2019-7-5)
289.Is the format correct? Give up; (2019-7-6)
290.----------------------------------------------------------------------------------------------------------------------
291.analyze model; data is all dinosaur’s name; what is {} in python; dictionary; char to x and x to char is dictionary; change the number of iterations; examples are shuffle of data; deal with vocab_size so that it can deal with different cases; let us set vecab size= 26; strange; change 26 to temp_size; (2019-7-6)
292.separate 27 out; delete the default expression; delete num_iteration as default; done; what is n_a; number of hidden unit; change n_a without default; n_a -- number of units of the RNN cell; done; delete dino_name as default; done; delete vocab_size as default; done; (2019-7-6)
293.The reason why we deal with dinosuar name is to get familiar with RNN model and see what we can do with similarity; delete learning_rate in default; done; it says we are out of bound; done; (2019-7-6)
294.Figure out meaning of X and Y below optimize; it is an example; add vocab size to optimize; should we take note when reading papers? Yes; (2019-7-7)
295.Can we find a space to show lexical result so that we can easily debug; change source code to preprocess result; can we come up with a way to easily debug translate_source? How php echo an array? Let us store source code in a txt file; create source code txt, done; change source code to preprocess result; give up; problem is solved; (2019-7-7)
296.Syntax analysis used result of preprocessor, rather than lexical analysis; combine == and != and < and ~; give up; it seems the symbol we send to html are IDList, how they become symbols? It is done through translate symbol; can we delete translate symbol and use php to implement it? (2019-7-8)
297.Delete translate symbol in insert_row; done; delete translate symbol in delete row; done; delete translate symbol in add element; done; the reason why we need translate is that we need translate symbol is to show import library; I think we do not need combine row after translation; give up delete translate symbol; (2019-7-8)
298.Pay attention to how we deal with reversed word in syntax analyze; reserved words contain word symbol made up with letters and phrase titles; (2019-7-8)
299.Separate syntax as generate symbols and generate IDList; done; change symbol list to message; hold; (2019-7-8)
300.----------------------------------------------------------------------------------------------------------------------
301.Deal with phrase; buy an mechanic keyboard; done; complement is C; (2019-7-8)
302.deal with similarity calculate, we need s(a, b) * s(b, a); distance between A to B and B to A should be equal, which is called symmetry; we should separate similarity out as a unique function; (2019-7-8)
303.Show select_sentence.php to show_select_sentence.php; done; change button name to show select sentence; done; separate print select sentence, we need it; we can copy it and then delete unnecessary ones; delete extras in print select sentence; done; (2019-7-8)
304.Separate similarity result as a single function; add conn->close at the end of each function; done; modify find_common.php so that the similarity is equal to sort_similar.php; delete extra {}; done; how should we deal with group? Keep it; delete extra element group from sort_similar.php; done; (2019-7-9)
305.Make find common to show correctly; natural language is right; check source code; find the reason; the separator becomes <br>; check object code; now we can represent source code correctly; the hard is that when compile, we consider all lines of source code as a whole, rather than line by line; done; (2019-7-9) 
306.Should we change similarity to distance? Hahahaha, cosine similarity is a kind of distance; check schema of Andrew Ng’s machine learning; give up; (2019-7-9)
307.利用相关系数解释相似度; now we know it is cosine similarity; (2018-10-25)
308.CS有点类似于汉明距离; now we know it is cosine similarity; (2019-7-9)
309.几何意义对于理解数学定理太重要了, 因为数学是数和形的科学。(2019-7-9)
310.----------------------------------------------------------------------------------------------------------------------
311.Modify find_common to take group into account; done; now they are equal; change split_elements_left to split_words_x; done; change split_elements_right to split_words_y; done; change quantity_left to quantity_x; done; change quantity_right to quantity_y; done; change element_frequency to word_frequency; done; change element_id to word_id; done; change element_name to word_symbol; done; (2019-7-9)
312.Chang element_group to word_group; done; change element_name to word_name; done; change num_element to num_word; done; change ElementID to WordID; done; (2019-7-9)
313.Modify random_select to show sentence correctly; done; (2019-7-9)
314.Should we separate print function of certain id, since in analyzer, most of buttons need this function; add object code for show sentence; separate for show_select_sentence.php; done; separate for find_similar.php; done; separate for find_common.php; done; (2019-7-10)
315.Change left_compare to x_compare; done; change right_compare to y_compare; done; change sql_left to sql_x; done; change sql_right to sql_y; done; change result_left to result_x; done; change result_right to result_y; done; change row_left to row_x; done; change row_right to row_y; done; change left_group to x_group; done; change right_group to y_group; done; change left to x; done; change right to y; done; (2019-7-10)
316.Deal with Laplace transform; change title of other Laplace transform; done; use import library to deal with other Laplace property; since there are more and more sentences in database, we can use analyzer to check the result; done; (2019-7-10)
317.Add phrases to SDB; delete preference; done; create table phrases; done; write a html to show all phrases; delete the strange button in phrases.html; done; add phrases tag between words and sentences; make them turn blue and white; done; add phrase tag in phrase; make them turn blue and white; done; add phrase tag in sentences; make them turn white and blue; done; add phrases in word_editor; make them turn white and blue; done; add phrases to sentence editor; make them turn white and blue; done; add phrases to analyzer; make them turn white and blue; done; (2019-7-10)
318.Deal with multiple reference; import library should be separated with semicolon; done; (2019-7-10) 
319.Check transform in database; keep it; change triangle to sine; add cosine into words; set position; done; (2019-7-10)
320.----------------------------------------------------------------------------------------------------------------------
321.Modify rand_select to use functions print_sentence; done; (2019-7-10)
322.Check papers received from Prof. He; check CA; done; check Latex/word; done; replace sentence that has only one line; 29; delayed unit step; 34th, nth power; add Factorial to phrases; 38, sin of Laplace; the reason why division by zero is that we should update abundance; done; (2019-7-10)
323.Change query_word_contained sentence to symbol; done; I think we should place id and type in the same table; modify print_sentence.php; done; (2019-7-10)
324.Separate e and pai, they are totally different; change cst into euler; done; add pai into words; done; (2019-7-10)
325.Add sinh and cosh into sentences; 40; sinh; add sinh to phrases; done; add cosh to phrases; done; 43, tanh; done; 44, coth; done; continue add laplace transform; done; (2019-7-11)
326.After complete introduction in algorithm, let’s collect items in SSFCDB255; (2019-7-11)
327.Place algorithm in introduction to algorithm in SDB; already exist, left-rotate; RB-insert; RB-transplant; bubble sort; RB-insert-fixup; RB-delete; tree-minimum; (2019-7-11)
328.Read content; done; read preface; foundations; introduction; the role of algorithms in computing; 1.1, algorithms; what kind of problems are solved by algorithms; data structures; technique; hard problems; Parallelism; 1.2, algorithms as a technology; efficiency; algorithms and other technologies; exercise; problems; (2019-7-11)
329.Getting started; 2.1, insertion sort; try to download office 2007; add < in the words; 73, insertion sort; change contain to less than; install office 2007; give up; us python to separate line; add greater than into words; change increase into greater than; done; add > in translate_source; (2019-7-11)
330.----------------------------------------------------------------------------------------------------------------------
331.combine === and != and < in an array; can we delete outsider objectcode? Yes; deal with ==; done; deal with <; done; deal with ~; done; add > in translate source; done; (2019-7-11)
332.Change sequence name to arr; done; exercise; analyzing algorithms; analysis of insertion sort; worst-case and average-case analysis; order of growth; exercises; designing algorithms; the divide-and-conquer approach; done; since in most cases, they use [] to represent array, this time we use [] to represent array; we should do this in php file; in preprocess; done; (2019-7-11)
333.Deal with 4 arguments; 79, merge-sort; deal with floor and ceiling; done; analyzing divide and conquer algorithms; analysis of merge sort; bubble sort; (2019-7-11)
334.Growth of functions; notation; O-notation; comparing functions; standard notations and common functions; polynomials; log; factorials; (2019-7-11)
335.Change element group to word group; change php_ElementGroup to WordGroup; done; change php_ElementName to WordName; done; change php_PhysicalAddress to WordID; done; change php_symbol to symbol; done; change php_POS to POS; done; change php_content to content; done; change php_comment to comment; (2019-7-12)
336.Change ElementName to WordName; done; change ElementSymbol to WordSymbol; done; change ElementPOS to WordPOS; done; change ElementContent to WordContent; done; change ElementComment to WordComment; done; (2019-7-12)
337.Divide and conquer; the maximum subarrary problem; add reference into print_sentence; done; add input and output into words; since we already have group, we can add word to vocabulary as many as possible; add lower bound and higher bound into words; add else into words; done; modify translate_source to deal with else; done; (2019-7-12)
338.Add title in similar_sentence; done; deal with square matrix; done; The substitution method for solving recurrence; the substitution method for solving recurrence; making a good guess; avoiding pitfalls; changing variables; the recurrence tree method; the master method for solving; the master theorem; using the master method; proof; floors and ceilings; (2019-7-12)
339.Change ElementGroup to WordGroup; done; change num_element to num_word; done; change elements to words; done; change compounds to sentences; done; (2019-7-12)
340.----------------------------------------------------------------------------------------------------------------------
341.Probabilistic Analysis and Randomized Algorithms; the hiring problem; worst case analysis; randomized algorithm; indicator random variables; randomized algorithms; randomly permuting arrays; add swap into words; Probabilistic analysis and further uses of indicator random variables; the birthday paradox; balls and bins; streaks; the on-line hiring problem; (2019-7-12)
342.We can conceal some details in the lectures; sorting and order statistics; introduction; the structure of data; why sorting; sorting algorithms; (2019-7-12)
343.heap sort; heaps; add two into words; done; should we take var into account? Let us try; 14, max-heapify; add left and right into words; done; change norm to size; done; 20 / page 157; change left and right to verb; done; (2019-7-13)
344.Example; Laplace; tanh; 利用sin和cos的对偶性; (2019-7-13)
345.Move update abundance in analyzer; done; replace compound with sentence; done; change num_element to num_words; done; the heap sort algorithm; priority queues; (2019-7-13)
346.Quick sort; description of quick sort; 109, page 171; performance of quicksort; worst-case; balanced; intuition; a randomized version of quicksort; add true and false to words; change By Group to Details; analysis of quick sort; expected running time; (2019-7-13)
347.sorting in linear time; lower bound for sorting; the decision tree model; counting sort; radix sort; (2019-7-13) 
348.median and order statistics; selection in expected linear time; selection in worst case; (2019-7-13)
349.Make PPT for our theory; what does word embedding belong to? NLP; finally, our theory is released from the shadow of TRIZ; 145, RB-delete; (2019-7-13)
350.----------------------------------------------------------------------------------------------------------------------
351.从几个例子中我们可以看到，结构相似的算法往往功能也相似；符号逻辑/形式逻辑；过去对算法的研究往往专注于功能，而没有专注于结构，或者说形式；(2019-7-13)
352.Introduction; elementary data structures; stacks and queues; 126, page 233; linked list; searching a linked list; add prev and next into word; done; implementing pointers and objects; representing rooted trees; (2019-7-14) 
353.Hash tables; hash tables; open addressing; add NIL into words; perfect hashing; (2019-7-14)
354.Binary search trees; what is binary search tree; what is a binary search tree; querying in a binary search tree; add convergence into words; 166, Page 292; randomly build binary search trees; (2019-7-14)
355.Red black trees; rotations; add right rotate into sentences; 171; 172, page 323; (2019-7-15)
356.Augmenting data structures; how to augment a data structure; interval trees; 175, interval search; (2019-7-15)
357.Advanced design and analysis techniques; introduction; (2019-7-15)
358.Dynamic programming; rod cutting; 177, more modify; matrix chain multiplication; 183, page 375; should we treat = as another way? Delete priority level 7, and modify generate symbol sequence; done; we should write a function that can compile all sources; find our graduate ppt; done; set title font size 36; learn to add pages to slides; done; add name to slides; element of dynamic programming; longest common sequence, 188; (2019-7-15)
359.Make indent shown correctly; done; optimal binary search trees; 190, optimal BST; backup database; change compound to sentence; hold; (2019-7-15)
360.----------------------------------------------------------------------------------------------------------------------
361.Add words into slides; done; greedy algorithm; elements of greedy strategy; greedy choice property; Huffman codes; done; separate union; done; (2019-7-16)
362.Amortized Analysis; aggregate analysis; the accounting method; (2019-7-16)
363.Advanced data structures; introduction; B-trees; swap 53 and 54; add recursion into words; 202, page 494; when object code is much larger than compare, it is useless; set to zero, like cut gradient; stop at 203; (2019-7-16)
364.Fibonacci Heaps; structure of heaps; heap operations; 213, cascading; (2019-7-16)
365.van Emde Boas Trees; 216, page 541; modify show_all_sentences.php so that it is shown more pretty; change similarity to numerator; done; make it more efficient; separate calculate similarity; input, two texts; output, similarity value; first deal with find_common.php; change similarity to numerator; done; add gradient cut in find common; done; delete num_row; done; delete source code; done; delete unnecessary elements; done; (2019-7-16)
366.Modify relation; done; data structures for DS; disjoint set forests; (2019-7-17)
367.Graph algorithms; introduction; Elementary Graph Algorithms; BFS; (2019-7-17)
368.Minimum Spanning Trees; (2019-7-17)
369.Single-Source Shortest Paths; (2019-7-17)
370.----------------------------------------------------------------------------------------------------------------------
371.All-Pairs Shortest Paths; (2019-7-17)
372.Maximum Flow; (2019-7-17)
373.Selected topic; Multithreaded Algorithms; 270, page 775; 280, page 803; (2019-7-17)
374.Matrix operations; 284, page 869; (2019-7-18)
375.Linear programming; ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
376.We need an id to text table; php 关联数组 need to be created; for find common, make print object code into a function; done; modify calculate similarity so that it is more efficient; done; (2019-7-18)
377.delete left and right in sort similarity; change row_left to row_x; done; change quantity left to quantity x; done; change split word left to split word x; done; change left_group to group_x; done; change right to y; done; (2019-7-18)
378.Add calculate similarity in sort similarity; hold; study php 关联数组; modify show all sentences; done; deal with && and ||; we should know more about assembly language; 使用尽可能少的寄存器; 289, page 917; (2019-7-20)
379.Search NLP conference on Zhihu; find tutorial books on Zhihu; (2019-7-19)
380.----------------------------------------------------------------------------------------------------------------------
381.与其他很多计算机分支方向类似，由于技术发展迅速，NLP/CL重视学术会议胜过学术期刊; 
382.Jurafsky和Martin的Speech and Language Processing
383.Chris Manning, introduction to information retrieval; (2019-7-19)
384.Download AI, a modern method; done; download network and crowds, done; (2019-7-20)
385.Download book of Qingcheng Zong; done; (2019-7-22)
386.Write weekly report; (2019-7-22)
387.Scientific research can be divided into applied research and basic research; Basic research is doing what you are curious about; Universities in the US do not do applied research, they only do basic research so as to focus on the next generation of talent; (2019-7-24)
388.How to pick a research problem? => What you are curious about? What do you want to do? What do you want to think 24 hours a day? (2019-7-24)
389.7:30, Room 103; (2019-7-24)
390.----------------------------------------------------------------------------------------------------------------------
391.Github get former version of program; done; change translate_source back so as to add = back; done; (2019-7-24)
392.买车票; 快递拆封; 取快递; contact with parents; (2019-7-20)
393.If you're going to use array_push() to insert a "$key" => "$value" pair into an array, it can be done using the following:
394.    $data[$key] = $value;
395.It is not necessary to use array_push.
396.We must have word group and frequency; two arrays, id_to_group and id_to_frequency; done; add them to calculate_similarity; done; change word_frequency to x_frequency; done; (2019-7-24)
397.Change word_to_group to id_to_group; done; replace x_group; done; replace y_group; done; add id_to_symbol; done; (2019-7-25)
398.Modify print_object_code in find_common, so as to use id_to_symbol; done; (2019-7-25)
399.Modify sort_similarity.php, so as to use id_to_group; add id to group and id to frequency; done; (2019-7-25)
400.----------------------------------------------------------------------------------------------------------------------
401.Separate confidence as a parameter; add id_to_group and id_to_frequency in calculate_all_similarity; change element_frequency to x_frequency; done; separate calculate similarity; (2019-7-25)
402.Separate object_code_x out; done; download 信息检索导论; done; (2019-7-29)
403.Write original and translation; done; (2019-7-30)
404.How to hide notes in wps? Done; (2019-7-31)
405.9天; 5天在; 4天离; 
406.150 *5/(5*3 + 4*2)
407.replace element with word; replace compound with sentence; done; (2019-8-9)
408.Replace database with knowledge base; (2019-8-9)
409.Read note of Wang Chao; 
410.----------------------------------------------------------------------------------------------------------------------
411.L2, supervised learning; L3, knn; 
412.We need a tutorial to Latex; install Miktex; done; open tex work; done; (2019-8-11)
413.First document; \commandname{option}; (2019-8-11)
414.02 sections; (2019-8-12)
415.03 packages; (2019-8-12)
416.04 math; (2019-8-12)
417.05 figures; (2019-8-12)
418.Package hyperref Error: Wrong driver option `xetex'
419.Download tex live; install tex live on windows; 
420.----------------------------------------------------------------------------------------------------------------------
421.Tex version; 
422.Package hyperref Error: Wrong driver option `xetex', (hyperref) because XeTeX is not detected.
423.Try to uninstall Miktex; done; restore system; done; (2019-8-14)
424.Try to download ctex; 
425.　TeX 系统是公认的数学公式排得最好的系统。美国数学学会（AMS） 鼓励数学家们使用 TeX 系统向它的期刊投稿。世界上许多一流的出版社如 Kluwer、Addison-Wesley、牛津大学出版社等也利用 TeX 系统出版书籍和期刊。
426.Leslie Lamport 开发的 LaTeX 是当今世界上最流行和使用最为广泛的 TeX 格式。它构筑在 Plain TeX 的基础之上，并加进了很多的功能以使得使用者可以更为方便的利用 TeX 的强大功能。使用 LaTeX 基本上不需要使用者自己设计命令和宏等，因为 LaTeX 已经替你做好了。因此，即使使用者并不是很了解 TeX ，也可以在短短的时间内生成高质量的文档。对于生成复杂的数学公式，LaTeX 表现的更为出色。
427.AMSTeX AMS-TeX 是美国数学会提供的，在 Plain TeX 基础上开发的 TeX 宏集。它主要用于排版含有很多数学符号和公式的科技类文章或书籍。AMSTeX AMS-TeX 给出了许多高级命令，可以让使用者很方便地排版大型的，复杂的数学公式。AMSTeX AMS-TeX 排版数学公式等的功能通过 AMSLaTeX AMS-LaTeX 中的宏包 amsmath 在 LaTeX 中得到实现。AMSTeX AMS-TeX 目前的版本为 2.1。
428.MiKTeX 是 Windows 下最好用的 TeX 系统。
429.TeXLive 是由国际 TeX 用户组织 TUG 开发的 TeX 系统，支持不同的操作系统平台。其 Windows 版本又称 fpTeX ， Unix/Linux 版本即著名的 teTeX 。
430.----------------------------------------------------------------------------------------------------------------------
431.CTeX 中文套装是基于 Windows 下的 MiKTeX 系统，集成了编辑器 WinEdt 和 PostScript 处理软件 Ghostscript 和 GSview 等主要工具。 CTeX 中文套装在 MiKTeX 的基础上增加了对中文的完整支持。 CTeX 中文套装支持 CJK, xeCJK, CCT, TY 等多种中文 TeX 处理方式。
432.Try to install xetex; done; use XeLatex to generate pdf file; done; (2019-8-15)
433.您的学号：M201973167
434.Select house;
435.青铜: 
436.西区小部分宿舍内部（图为西八舍）; 地理位置: 百景园; 
437.白银: 
438.西六舍; 地理位置; 西一学生食堂; 
439.南二舍; 光谷体育馆; 
440.----------------------------------------------------------------------------------------------------------------------
441.东四舍; 东一食堂; 
442.东七舍; 东一食堂; 
443.教七舍; 中部; 
444.黄金;
445.紫松; 最西; 
446.Change tags of notes; done; add math travel to github; done; (2019-8-15)
447.Read Andrew Ng’s kNN; done; (2019-8-16)
448.Find keywords in student’s notes; 
449.Assumption: data points that are similar have similar labels;
450.----------------------------------------------------------------------------------------------------------------------
451.???????
452.Distance is Minkowski distance; 
453.Find note of cornell; done; (2019-8-16)
454.Find how to write subscript; 
455.It is a way to specify constraints. To put it very simply, the problem "do 'X' subject to 'Y'" means that, you have to do "X" (whatever X is), but you have to do it such that "Y" is also satisfied in the process.
456.How can I get text underneath an underbrace?
457.latex \mathbb
458.How to write dot in Latex; \cdot; (2019-8-17)
459.Delete the previous chapter; done; (2019-8-17)
460.----------------------------------------------------------------------------------------------------------------------
461.Install wamp on windows; done; install vscode; done; (2019-8-17)
462.Install github desktop; done; install git on windows; done; (2019-8-17)
463.Use tex in the web to replace our text; done; (2019-8-17)
464.Learn how to load figure in Latex; done; (2019-8-18)
465.Learn how to change figure size in Latex; \includegraphics[width=0.5\linewidth]{manifold.png}; (2019-8-18)
466.Set picture here; done; (2019-8-18)
467.Learn how to place figure in the middle; done; (2019-8-18)
468.check whether all content on the student’s note shown on the lecture; done; (2019-8-18)
469.Check for error and format; done; (2019-8-18)
470.----------------------------------------------------------------------------------------------------------------------
471.证件: 身份证包包; 录取通知书档案夹; 
472.电子产品: 电脑鼠标电源线; 剃须刀; 充电宝; 手机充电器; 手机充电线; 电动牙刷;   
473.杂物: 钥匙; 水壶; 眼镜盒; 卫生纸; 论语孟子礼记;
474.Install wangyiyouxiangdashi; done; write weekly report; (2019-8-20)
475.Download wechat desktop; done; (2019-8-20)
476.Deal with school net; 
477.Check how to write paper with Latex; give up, since they do not have the same answer; (2019-8-21)
478.和计算机其他门类一样，AI不是用来学的，而是用来玩的，用来解决实际问题的，然后在解决实际问题的过程中深化对AI的理解; Just as other branches of computer, AI is used to play and solve practical problems rather than to learn. And in the process of solving practical problems, we get a deeper understanding of AI. (2019-7-5)
479.Andrew Ng告诉我们，一些看似简单的算法可能异常有效; According to Andrew Ng, some simple algorithms prove to be very useful in some cases. (2019-7-15)
480.----------------------------------------------------------------------------------------------------------------------
481.再开一个档案保管证明; give up, since they refused to do so；(2019-8-21)
482.Deal with import library; done; (2019-8-21)
483.Try to change color in input text box; give up, I cannot figure out why this is needed; (2019-8-21)
484.x + diff(y) == 4/9; 6.0 + diff(x*5 + 4); (2019-8-21)
485.Sorkine, O. and Alexa, M., 2007, July. As-rigid-as-possible surface modeling. In Symposium on Geometry processing (Vol. 4).
486.Deal with the problem of local define function; done; (2019-8-22)
487.Alan V. Oppenheim. (1999). Discrete-time signal processing (2nd ed). Prentice-Hall, Inc. p.11 ; (2018-1-16)
488.Check compounds concerning algorithm; cannot figure out the meaning; (2019-8-22)
489.就像散列算法应尽量简单一样, 比较相似度算法也应该尽量简单, 能够在较短的时间内计算出与所有条目相比的相似度; 没听说过; (2019-8-22)
490.----------------------------------------------------------------------------------------------------------------------
491.不同学科的知识在原生态共通点很多，就像细胞分化前很相似, 分化后就越差越远了; 因此, 可以更多关注原生态的知识; (2019-8-22)
492.Backup databases; done; (2019-8-22)
493.李开复：一个产业如果有大量的数据、数据带有标签而且是单一领域的，那就是人工智能应用最理想的数据。(2019-8-21)
494.Check wiki about knowledge engineering; expert systems; (2019-8-22) 
495.Check zhihu about expert system; check zhihu about knowledge engineering; search about knowledge graph; 
496.专家系统的核心是推理机。专家系统由事实数据库与推理规则组成，“专家”的意思是它只能解决某一领域的问题，运作原理就是通过推理规则对事实进行组合，得出结论。(2019-8-22)
497.专家系统又称符号主义 基于概率的又称是连接主义 ，专家系统目前价值很大 金融风控类应用目前还是比较典型的使用领域；专家系统核心是规则 而规则强依赖于领域内专家。(2019-8-22)
498.和其他学科不同, 数学各个门类之间联系非常紧密, 是指可以用盘根错节形容(解析几何, 代数拓扑); (2018-3-1)
499.Modify theory.html so that // can be good; I do not know its meaning; (2019-8-22)
500.----------------------------------------------------------------------------------------------------------------------
